components/WalletConnector/WalletConnector.tsx

```diff
@@ -0,0 +1,77 @@
+import QRCodeModal from '@walletconnect/qrcode-modal';
+import WalletConnect from '@walletconnect/client';
+import { useContext } from 'react';
+import { WalletAddressContext } from '../../contexts/WalletAddressContext';
+import { useRouter } from 'next/router';
+
+const WalletConnector = () => {
+  const { setWalletAddress } = useContext(WalletAddressContext);
+  const router = useRouter();
+  const showModal = async () => {
+    const connector = new WalletConnect({
+      bridge: 'https://bridge.walletconnect.org', // Required
+      qrcodeModal: QRCodeModal,
+    });
+
+    // Check if connection is already established
+    if (!connector.connected) {
+      // create new session
+      connector.createSession();
+    }
+
+    // Subscribe to connection events
+    connector.on('connect', (error, payload) => {
+      if (error) {
+        throw error;
+      }
+
+      // Get provided accounts and chainId
+      const { accounts, chainId } = payload.params[0];
+      if (chainId != 2828) {
+        alert('Please switch to L16 network');
+        return;
+      }
+      setWalletAddress(accounts[0]);
+      router.push({ pathname: `/${accounts[0]}/overview` });
+    });
+
+    connector.on('session_update', (error, payload) => {
+      if (error) {
+        throw error;
+      }
+
+      // Get updated accounts and chainId
+      const { accounts, chainId } = payload.params[0];
+      if (chainId != 2828) {
```

Comment:
see previous comment. You reuse that same number as in line 30.

If tomorrow it changes to 56, you have to update it everywhere in your code. Use a variable.